vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -665,10 +714,45 @@ def select_experts(hidden_states: torch.Tensor,
 
         return topk_weights, topk_ids
 
+    def naive_multicast(self, x: torch.Tensor,
+                        cu_tokens_across_dp_cpu: torch.Tensor):
+        assert (len(x.shape) == 2)
+        buffer = torch.empty((cu_tokens_across_dp_cpu[-1], x.size(1)),
```

Comment:
ideally this buffer can be static (with max size). and we can slice `buffer[:cu_tokens_across_dp_cpu[-1]]`.

Reply:
Any good ideas for where to allocate this buffer if we make it static?

Reply:
we can leave it as future work. I think we should also do chunking here. allocating a max-size buffer would be too crazy.

so I'm fine with the current status.